harness/src/main/java/org/apache/geode/perftest/yardstick/YardstickTask.java

```diff
@@ -107,6 +112,35 @@ public String defaultDescription() {
 
     runner.runBenchmark();
 
-    testDoneProbe.await();
+    Histogram lastHistogram = hdrHistogramProbe.getHistogram();
+    while (!testDoneProbe.await(10, TimeUnit.SECONDS)) {
+      boolean warmupFinished = hdrHistogramProbe.isWarmupFinished();
+      // Create a histogram for the previous 10 second window
+      Histogram histogram = hdrHistogramProbe.getHistogram();
+      Histogram currentHistogram = histogram.copy();
+      if (histogram.getStartTimeStamp() == lastHistogram.getStartTimeStamp()) {
+        currentHistogram.subtract(lastHistogram);
+        currentHistogram.setStartTimeStamp(lastHistogram.getEndTimeStamp());
+      }
+
+      String prefix = warmupFinished ? "WORK" : "WARMUP";
+
+      // Log the histogram
+      logProgress(context, prefix, currentHistogram);
+
+      // Capture the current histogram
+      lastHistogram = histogram;
+    }
+    logProgress(context, "TOTAL", hdrHistogramProbe.getHistogram());
```

Comment:
Why don't tasks just invoke the logger directly? What is purpose of abstracting loging into this `logProgress` method on each task?

Reply:
This logProgress is sending the log to the controller JVM. Those are the only logs that go to the console. The individual worker JVMs are just logging to their own log files.